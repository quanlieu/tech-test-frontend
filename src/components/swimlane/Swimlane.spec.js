import React from 'react'
import { render } from '@testing-library/react'
import { Swimlane } from './Swimlane'

describe('Swimlane', () => {
  it('renders', async () => {
    const lanes = [
      {
        title: "Sam Seaborn",
        cards: [
          {
            style: { backgroundColor: "palegreen", color: "darkgreen" },
            className: "cardContent",
            description: "Shield some wiring",
            start: new Date("2018-09-01T09:00:00.000Z"),
            end: new Date("2018-09-01T13:00:00.000Z"),
          },
        ],
      },
      {
        title: "Donna Moss",
        cards: [
          {
            style: { backgroundColor: "lightsalmon", color: "darkred" },
            className: "cardContent",
            description: "Meal Break",
            start: new Date("2018-09-01T12:15:00.000Z"),
            end: new Date("2018-09-01T13:10:00.000Z"),
          },
          {
            style: { backgroundColor: "palegreen", color: "darkgreen" },
            className: "cardContent",
            description: "Build a shed",
            start: new Date("2018-09-01T11:00:00.000Z"),
            end: new Date("2018-09-01T10:15:00.000Z"),
          },
        ],
      },
      {
        title: "Toby Ziegler",
        cards: [
          {
            description: "Meal Break",
            start: new Date("2018-09-01T12:15:00.000Z"),
            end: new Date("2018-09-01T13:10:00.000Z"),
          },
        ],
      },
    ]
    expect(render(
      <Swimlane 
        start={new Date('2018-09-01T00:00:00Z')}
        end={new Date('2018-09-01T24:00:00Z')}
        lanes={lanes}
      />
    )).toMatchSnapshot()
  })
});
