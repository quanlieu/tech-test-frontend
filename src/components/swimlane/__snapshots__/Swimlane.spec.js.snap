// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Swimlane renders 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="swimlane__container"
      >
        <div
          class="swimlane__lane"
        >
          <div
            class="swimlane__title"
          >
            Sam Seaborn
          </div>
          <div
            class="swimlane__detail"
          >
            <div
              class="swimlane__card cardContent"
              style="background-color: palegreen; color: darkgreen; left: 37.5%; width: 16.666666666666668%;"
            >
              Shield some wiring
            </div>
          </div>
        </div>
        <div
          class="swimlane__lane"
        >
          <div
            class="swimlane__title"
          >
            Donna Moss
          </div>
          <div
            class="swimlane__detail"
          >
            <div
              class="swimlane__card cardContent"
              style="background-color: lightsalmon; color: darkred; left: 51.041666666666664%; width: 3.8194444444444446%;"
            >
              Meal Break
            </div>
          </div>
        </div>
        <div
          class="swimlane__lane"
        >
          <div
            class="swimlane__title"
          >
            Toby Ziegler
          </div>
          <div
            class="swimlane__detail"
          >
            <div
              class="swimlane__card "
              style="left: 51.041666666666664%; width: 3.8194444444444446%;"
            >
              Meal Break
            </div>
          </div>
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="swimlane__container"
    >
      <div
        class="swimlane__lane"
      >
        <div
          class="swimlane__title"
        >
          Sam Seaborn
        </div>
        <div
          class="swimlane__detail"
        >
          <div
            class="swimlane__card cardContent"
            style="background-color: palegreen; color: darkgreen; left: 37.5%; width: 16.666666666666668%;"
          >
            Shield some wiring
          </div>
        </div>
      </div>
      <div
        class="swimlane__lane"
      >
        <div
          class="swimlane__title"
        >
          Donna Moss
        </div>
        <div
          class="swimlane__detail"
        >
          <div
            class="swimlane__card cardContent"
            style="background-color: lightsalmon; color: darkred; left: 51.041666666666664%; width: 3.8194444444444446%;"
          >
            Meal Break
          </div>
        </div>
      </div>
      <div
        class="swimlane__lane"
      >
        <div
          class="swimlane__title"
        >
          Toby Ziegler
        </div>
        <div
          class="swimlane__detail"
        >
          <div
            class="swimlane__card "
            style="left: 51.041666666666664%; width: 3.8194444444444446%;"
          >
            Meal Break
          </div>
        </div>
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
